exporter.py

```diff
@@ -54,12 +55,41 @@ def parse_params():
 
 def run_prometheus_server(port, collectors, *args):
     start_http_server(int(port))
+    collector_list = list()
     for c in collectors:
         REGISTRY.register(c)
+        collector_list.append(REGISTRY._get_names(c))
+
+    metrics = get_collector_up_information(port)
+    REGISTRY.register(CollectorUp(collector_list, metrics))
 
     while True:
         time.sleep(1)
 
+def get_collector_up_information(port):
```

Comment:
Couldn't that be in your collector? If you make sure, it gets registered after the other collectors that local connection would return the other metrics too.

Reply:
Yes indeed, I'm about to change it.